Document how to localized code strings of type LocalizedStringResource (#266)
FlineDev/BartyCrouch: README.md

- `localizablePaths`: The enclosing path(s) containing the localized `Localizable.strings` files.
- `defaultToKeys`: Add new keys both as key and value.
- `additive`: Prevents cleaning up keys not found in code.

> - `customFunction`: Use alternative name to search for strings to localize, in addition to `NSLocalizedString`, and `CFCopyLocalizedString`. If not specified, look for `LocalizedStringResource`.

Reviewer: This could a bit shorter. For example:
```suggestion
- `customFunction`: Use alternative name to search for strings to localize, in addition to `NSLocalizedString`, and `CFCopyLocalizedString`. Defaults to `LocalizedStringResource`.
```

NOTE: As of version 4.x of BartyCrouch *formatted* localized Strings are not supported by this automatic feature.

> ### Localizing strings of `LocalizableStringResource` type (AppIntents, SwiftUI, ...)

Reviewer: Note sure if `LocalizedStringResource` is used in SwiftUI, you might mix it up with `LocalizedStringKey`...

Historically, Apple platforms used `CFCopyLocalizedString`, and `NSLocalizedString` macros and their variants, to mark strings used in code to be localized, and to load their localized versions during runtime from `Localizable.strings` file.

> Since introduction of SwiftUI, the localized strings in code are typically typed as `LocalizedStringResource`, and are no longer marked explicitely.

Reviewer: Aren't SwiftUI views localized via `LocalizedStringKey`? As far as I know, only few frameworks make use of `LocalizedStringResource` and unlike `LocalizedStringKey`, the `LocalizedStringResource` type isn't defined in the SwiftUI framework but [in Foundation](https://developer.apple.com/documentation/foundation/localizedstringresource). Therefore I find this sentence misleading, but maybe I'm missing something. I would change as follows:
```suggestion
Since introduction of the AppIntents framework, the localized strings in code can also be typed as `LocalizedStringResource`, and are no longer marked explicitly.
```
